adapters/webhdfs-adapter/src/main/java/com/connexta/replication/adapters/webhdfs/WebHdfsNodeAdapter.java

```diff
@@ -382,9 +384,21 @@ String getVersion4Uuid(String fileUrl, Date modificationTime) {
     return addFailedItemsToRelevantFiles(results, oldFiles, failedItemIds);
   }
 
-  private List<FileStatus> addFailedItemsToRelevantFiles(List<FileStatus> relevantFiles, Map<String, FileStatus> oldFiles, List<String> failedItemIds) {
+  /**
+   * Iterates through the list of failed IDs to look up the associated {@link FileStatus} in the
+   * map.
+   *
+   * @param relevantFiles - the list of {@link FileStatus} to be returned
+   * @param oldFiles - the list of {@link FileStatus} with a modification date after the filter date
+   * @param failedItemIds - the list of failed item IDs
+   * @return A list of {@link FileStatus} representing the full set of files to be returned
+   */
+  private List<FileStatus> addFailedItemsToRelevantFiles(
+      List<FileStatus> relevantFiles,
+      Map<String, FileStatus> oldFiles,
+      List<String> failedItemIds) {
     // only need to do the lookup if there are items in the failed ID list
```

Comment:
If you want to keep this comment, you should add the criteria that old files are also required for a lookup to occur.